homeassistant/components/tcp/common.py

```diff
@@ -99,6 +99,9 @@ def name(self) -> str:
 
     def update(self) -> None:
         """Get the latest value for this sensor."""
+        if self._state is not None:
+            # do not set it as unavailable if still unknown
```

Comment:
This comment says to not set it as unavailable but just below it actually does that.
I assume setting it as unavailable is the purpose so please update the comment to reflect what actually going to happen.

Reply:
Won't reading the whole comment (instead of half) make it clearer?
Do not set to unavailable _if unknown_
Goal being to retain the unknown initial state

Reply:
The comment is a bit confusing so maybe as you say, put it above the `if` and `# Don't set unavailable if state exist` or something like that.